### PR TITLE
Document the -native-compiler option

### LINKED ISSUE #12564: Documentation of native flags
It seems to me that there is no specification of how native flags should work; this makes it hard to support native properly as we have to guess what the intended behavior is, and this may not match implemented one.

For now we have two flags:

- `configure`'s `-native-compile={yes,no}`
- `coqc`'s `-native-compiler={yes,no,ondemand}`

that gives us 6 possible combinations to document. In particular, important points for each combination (set) are:
- files that are produced
- files that are required

### PR DESCRIPTION
**Pull request by proux01**
**Kind:** documentation.

Fixes #12564

### CODE REVIEW

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (silene) [inline doc/sphinx/practical-tools/coq-commands.rst:226]: I would split this sentence into two parts, to make it a bit less confusing: "Setting `yes` enables `native_compute`; it also causes Coq to precompile the native code for future uses; all dependencies need to have been precompiled beforehand. Setting `no` disables `native_compute`; no files are precompiled.

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (silene) [inline doc/sphinx/practical-tools/coq-commands.rst:230]: "calling". Also, rather than "temporarily", I would write something like "All missing dependencies will be recompiled every time `native_compute` is called."

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (silene) [inline doc/sphinx/practical-tools/coq-commands.rst:273]: "aforementioned"

File: doc/sphinx/proofs/writing-proofs/rewriting.rst
Comment (silene) [inline doc/sphinx/proofs/writing-proofs/rewriting.rst:446]: I would just remove "a lot of".

File: doc/sphinx/proofs/writing-proofs/rewriting.rst
Comment (silene) [inline doc/sphinx/proofs/writing-proofs/rewriting.rst:447]: "precompiled dependencies" seems better.

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (silene) [inline doc/sphinx/practical-tools/coq-commands.rst:228]: I would just remove "itself".

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (Zimmi48) [inline doc/sphinx/practical-tools/coq-commands.rst:231]: Up to you, but you could use the `.. versionchanged::` Sphinx directive here. Its usage is the same as for `.. deprecated::`.

**Comment by Zimmi48** [comment]
@silene Would you like to be the assignee for this PR?

File: doc/sphinx/proofs/writing-proofs/rewriting.rst
Comment (jfehrle) [inline doc/sphinx/proofs/writing-proofs/rewriting.rst:445]:
```suggestion
   computations. With this configuration, this tactic can default to
```

"According to the configuration" doesn't sound very good.  Not sure if my suggestion means what you intend.

**Review by jfehrle — COMMENTED** [review]
minor comments

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (silene) [reply doc/sphinx/practical-tools/coq-commands.rst:273]: What the... How are users even supposed to guess that this button means "insert a suggestion"?! Thanks a lot.

File: doc/sphinx/proofs/writing-proofs/rewriting.rst
Comment (silene) [reply doc/sphinx/proofs/writing-proofs/rewriting.rst:445]: Not quite. "Depending on" would fit better the intended meaning. I guess "either" should also be added before "default".

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (jfehrle) [reply doc/sphinx/practical-tools/coq-commands.rst:273]: If you position the cursor over the icon it shows the tooltip "Insert a suggestion".  This is a pretty standard idiom for GUIs.

Comment (proux01) [reply doc/sphinx/practical-tools/coq-commands.rst]: (it also took me some hovering to find that button)

**Comment by proux01** [comment]
Thanks for the relevant reviews! They are taken into account.

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (silene) [inline doc/sphinx/practical-tools/coq-commands.rst:251]: I do not understand what "temporary" means here.

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (silene) [inline doc/sphinx/practical-tools/coq-commands.rst:262]: Is that true? (I did not check.) I naively would have expected an error for `yes`.

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (silene) [inline doc/sphinx/practical-tools/coq-commands.rst:259]: I do not think the user is interested by a "temporary" outcome, so it might be better to just write "none" here.

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (silene) [inline doc/sphinx/practical-tools/coq-commands.rst:248]: Is it worth adding a "`native_compute`" column, so that we can write "yes" or "`vm_compute`"?

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (proux01) [reply doc/sphinx/practical-tools/coq-commands.rst]: indeed, removed

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (proux01) [reply doc/sphinx/practical-tools/coq-commands.rst:265]: When native_compute is disabled at configure time, the `-native-compiler` option is indeed just ignored (with a warning).

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (proux01) [reply doc/sphinx/practical-tools/coq-commands.rst]: agree

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (proux01) [reply doc/sphinx/practical-tools/coq-commands.rst:248]: Indeed, that's much clearer than those "temporary".

**Comment by silene** [comment]
Fine with me. Once it has been squashed, I will merge it, if there are no other comments by then.

Here is a link to the [rendered documentation](https://coq.gitlab.io/-/coq/-/jobs/936067352/artifacts/_install_ci/share/doc/coq/sphinx/html/practical-tools/coq-commands.html#by-command-line-options), in case someone wonders what it looks like.

**Comment by proux01** [comment]
Squashed, CI green (compcert failure is definitely unrelated)

**Review by silene — APPROVED** [review]

**Comment by silene** [comment]
@coqbot merge now

File: doc/sphinx/practical-tools/coq-commands.rst
Comment (Blaisorblade) [inline doc/sphinx/practical-tools/coq-commands.rst:231]: - Some readers might not realize this "recompilation" has no caching.
- Does "every time" mean that `native_compute. native_compute.` would recompile dependencies twice?
If not, I'd recommend a different text — "missing dependencies will be recompiled as needed for files :tacn:`native_compute` is called`. I thought they'd be cached within the same file, but I'd rather not guess.

Comment (Blaisorblade) [reply doc/sphinx/practical-tools/coq-commands.rst:231]: - Skimming the review comments, I think the "temporary" recompilation went in the right direction...
- Being too specific on `ondemand` might not be worth it.

**Pull request merged**
